CollapseLauncher/XAMLs/StartUp/StartupLanguageSelect.xaml.cs

```diff
@@ -31,12 +34,86 @@ public StartupLanguageSelect()
             }
         }
 
-        private void ComboBox_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        private string LanguageFallback(string tag)
         {
-            string LangID = LanguageIDIndex[(sender as ComboBox).SelectedIndex];
-            SetAppConfigValue("AppLanguage", LangID);
-            LoadLocale(LangID);
-            NextBtn.IsEnabled = true;
+            tag = tag.ToLower();
```

Comment:
This can be simplified to a switch-case statement instead of multiple ifs. Also, you can trim the string to only keep the first 2 or 3 letters.

Reply:
Okay. but still need additional process for Chinese and Portuguese, because we have multiple variants in Collapse.